### PR TITLE
feat: drawing layer

### PR DESCRIPTION
**Pull request by buenaflor**
Adds a drawing layer.

Features:
 - Zooming
 - Dragging
 - Scrolling
 - Single select & multi select of shapes
 - Resizing & rotation of shapes
   -  including a group of selected shapes at once

Controls for mouse:
 - Dragging/moving the canvas -> Wheel button
 - Scrolling -> Wheel scroll
 - Zooming -> Ctrl + Wheel scroll
 - Selection -> Either single select a shape by clicking the shape once or multi select by holding mouse button and going over the desired shapes

Controls for touchpad (not tablets):
 - Dragging/moving, scrolling the canvas -> Normal swiping
 - Zooming: same gesture as zooming in on phones and tablets
 - Selection stays the same

It's still WIP but if possible test out current features.

There are a couple minor things that need fixing that influence UX:
 - Scaling factor in regards to zooming and scrolling should be different from mouse and touchpad users
 - Selected shape group can only be moved around when clicking a shape and not the selection rectangle

This was honestly quite difficult so it might take a bit of time to make this experience more smooth.

## Basics

- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: frontend/src/routes/index.ts [hunk 1]
```diff
@@ -23,6 +25,11 @@ const routes: Routes = {
     path: '/',
     title: 'PermaPlanT',
   },
+  [Pages.DrawingPage]: {
+    component: DrawingPage,
+    path: '/drawing-page',
```
Comment (markus2330) [inline frontend/src/routes/index.ts:30]: And please also add a link from landing page (rebase first):

```suggestion
    path: '/map',
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Amazing job, we love this first version from UI perspective!

Important stuff first, to get it merged:

- [ ] write a README.md how to add new layers (docu for other devs)
- [ ] how to reuse common parts (e.g. selection will be used everywhere)
- [ ] write a README explaining how to use the UI (more or less what is written here in the PR description)
- [ ] add more code documentation

From UI perspective (can also be fixed in a later PR, please create an issue if you decide to do so):

- [ ] group selection: selection disappears after objects were moved
- [ ] ctrl should allow multi-selection of different objects (also expanding on group selection)

**Comment by markus2330** [comment]
@Benjaminimal @badnames as discussed, please review this while starting to implement your own layer. If you feel confident we should merge this, please approve.

**Review by Benjaminimal — APPROVED** [review]
Looks very usable.
The next step would be those side panes with tools and such.
Do we have an issue for that yet?

**Comment by markus2330** [comment]
> Looks very usable.

Thanks for the review! @badnames any further comments?

> The next step would be those side panes with tools and such.
> Do we have an issue for that yet?


We already started with the implementation: #244

**Comment by Benjaminimal** [comment]
Would love to get this merged soon so I'm going to hijack this and fix the merge conflicts.

**Comment by markus2330** [comment]
Yes, please go ahead :wink:

**Comment by Benjaminimal** [comment]
jenkins build please

**Comment by Benjaminimal** [comment]
@markus2330 
@lukashartl 
Seems like we have some issues installing npm packages on our Jenkins server.
Can you please help?
https://build.libelektra.org/blue/organizations/jenkins/PermaplanT/detail/PR-215/5/pipeline/138

**Comment by Benjaminimal** [comment]
jenkins build please

**Comment by Benjaminimal** [comment]
Installing npm packages works again.
I'll fix the remaining issue.

**Comment by markus2330** [comment]
release was done, so we can merge it now!

**Pull request merged**